walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,309 @@
+    Collections.sort(filteredEvents, (event1, event2) -> TimeRange.ORDER_BY_START.compare(event1.getWhen(), event2.getWhen()));
+
+    List<TimeRange> openTimes = new ArrayList<TimeRange>();
+    int numEvents = filteredEvents.size();
+
+    // Move through sorted events looking for large enough gap between consecutive events
+    for (int i = 0; i < numEvents - 1; i++) {
+        // Generate a new event for the second event taking into account overlap compensation (look at function for more info)
+        String title = filteredEvents.get(i + 1).getTitle();
+        TimeRange compensatedTimeRange = compensateOverlap(filteredEvents.get(i).getWhen(), filteredEvents.get(i+1).getWhen());
+        Collection attendees = filteredEvents.get(i + 1).getAttendees();
+        Event newTiming2 = new Event(title, compensatedTimeRange, attendees);
+
+        // Update second event, calculate difference, then include if open slot is long enough
+        filteredEvents.set(i + 1, newTiming2); 
+        ValidOpenTimeRange dif = calculateEventDif(filteredEvents.get(i).getWhen(), filteredEvents.get(i+1).getWhen(), neededDif);
+        if (dif.checkDuration()) { openTimes.add(dif.getOpenTime()); }
+    }
+
+    return openTimes;
+  }
+
+  /**
+   * Create new list of events (only those attendees by the provided attendees)
+   *
+   * @param events          A collection of all the Event objects
+   * @param attendees       Set of relevant attendees (only care about their events)
+   */
+  private List<Event> filterEvents(Collection<Event> events, Collection<String> attendees) {
+      List<Event> filteredEvents = new ArrayList<Event>();
+      Set<String> requiredAttendees = new HashSet<String>(attendees);
+
+      // Check if any attendee of interested attended given event -> if so add to filtered events
+      for (Event event : events) {
+        Set<String> eventAttendees = event.getAttendees();
+        Set<String> intersection = Sets.intersection(eventAttendees, requiredAttendees);
+
+        if (containsRequiredAttendees(intersection)) { filteredEvents.add(event); }
+      }
+
+      return filteredEvents;
+  }
+
+  /**
+   * Event contains required attendees if the size of intersection > 0
+   *
+   * @param intersection    Intersection of attendees of interest and Event's attendees
+   */
+  private boolean containsRequiredAttendees(Set<String> intersection) {
+      return intersection.size() > 0;
+  }
+
+  /**
+   * Calculates the different between consecutive events and whether it is sufficiently long
+   * to satisfy the meeting request.
+   *
+   * @param timing1     The TimeRange of the first event
+   * @param timing2     The TimeRange of the second event
+   * @param neededDif   The time gap necessary to schedule the request
+   * 
+   */
+  private ValidOpenTimeRange calculateEventDif(TimeRange timing1, TimeRange timing2, long neededDif) {
+      // Check for overlap btw ranges -> no time in between
```

Comment:
btw should become between (readability is more important that the length of comments)
Also, i'm not sure i like the `->` arrows. They won't match anyone else's comments and don't really add to the comment.

Reply:
sounds good!